src/runmode-dpdk.c

```diff
@@ -0,0 +1,1274 @@
+        SCLogError(SC_ERR_INVALID_VALUE, "Invalid number of queues (max %" PRIu32 ")",
+                RTE_MAX_QUEUES_PER_PORT);
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    SCReturnInt(0);
+}
+
+int ConfigSetTxQueues(DPDKIfaceConfig *iconf, const char *entry_str)
+{
+    if (entry_str == NULL) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Passed value is NULL");
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    if (strcmp(entry_str, "auto") == 0) {
+        iconf->nb_tx_queues = iconf->threads;
+        SCReturnInt(0);
+    }
+
+    if (StringParseUint16(&iconf->nb_tx_queues, 10, 0, entry_str) < 0) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Value %s can not be parsed to int", entry_str);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    if (iconf->nb_tx_queues < 1 || iconf->nb_tx_queues > RTE_MAX_QUEUES_PER_PORT) {
+        SCLogError(SC_ERR_INVALID_VALUE,
+                "Invalid number of queues "
+                "(max %" PRIu32 ")",
+                RTE_MAX_QUEUES_PER_PORT);
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    SCReturnInt(0);
+}
+
+int ConfigSetMempoolSize(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Invalid memory pool size");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->mempool_size = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetMempoolCacheSize(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0 || entry_int > RTE_MEMPOOL_CACHE_MAX_SIZE) {
+        SCLogError(SC_ERR_INVALID_VALUE,
+                "Invalid memory pool cache size"
+                "(max %" PRIu32,
+                RTE_MEMPOOL_CACHE_MAX_SIZE);
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->mempool_cache_size = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetRxDescriptors(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Number of RX descriptors must be a positive number");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->nb_rx_desc = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetTxDescriptors(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Number of TX descriptors must be a positive number");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->nb_rx_desc = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetMtu(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int < RTE_ETHER_MIN_MTU || entry_int > RTE_ETHER_MAX_JUMBO_FRAME_LEN) {
+        SCLogError(SC_ERR_DPDK_CONF, "Size of MTU must be between %" PRIu32 " and %" PRIu32,
+                RTE_ETHER_MIN_MTU, RTE_ETHER_MAX_JUMBO_FRAME_LEN);
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->mtu = entry_int;
```

Comment:
please refer to comment shared in YAML section about MTU and rework

Reply:
I replied to MTU comment earlier you there is confusion is the exact working of DPDK and requirement on Suricata